crypto/fips1402/src/main/java/org/keycloak/crypto/fips/FIPS1402Provider.java

```diff
@@ -262,4 +259,34 @@ private SNIHostName getSNIHostName(String host) {
 
         };
     }
+
+    // BCFIPS require "SecureRandom.getInstanceStrong" to be available. But it may not be available on RHEL 8 on OpenJDK 17 due the https://bugzilla.redhat.com/show_bug.cgi?id=2155060
```

Comment:
I'm wondering if this is actually a bug or behavior we need to take into account when running on RHEL. See https://access.redhat.com/documentation/en-us/openjdk/17/pdf/configuring_openjdk_17_on_rhel_with_fips/openjdk-17-configuring_openjdk_17_on_rhel_with_fips-en-us.pdf.

I'm not sure about this, but looks like you can use `java.security.disableSystemPropertiesFile` to disable global FIPS and then force a specific algorithm.

Reply:
The `java.security.disableSystemPropertiesFile` is applied at bootstrap of JVM and it completely disables FIPS for the JDK (Similar to `com.redhat.fips`). Using this property is not an option as we won't be FIPS compliant when it is used.